startup/GafferScene/renderPassTypes.py

```diff
@@ -0,0 +1,520 @@
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are
+#  met:
+#
+#      * Redistributions of source code must retain the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer.
+#
+#      * Redistributions in binary form must reproduce the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer in the documentation and/or other materials provided with
+#        the distribution.
+#
+#      * Neither the name of John Haddon nor the names of
+#        any other contributors to this software may be used to endorse or
+#        promote products derived from this software without specific prior
+#        written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+#  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+#  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+#  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+#  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+#  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+#  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+##########################################################################
+
+import imath
+import inspect
+
+import IECore
+import IECoreScene
+
+import Gaffer
+import GafferScene
+
+def __renderPassShaderAssignment( usage ):
+
+	processor = GafferScene.SceneProcessor()
+
+	processor["renderer"] = Gaffer.StringPlug()
+
+	processor["defaultShader"] = Gaffer.ObjectPlug( defaultValue = IECore.NullObject() )
+	processor["defaultArnoldShader"] = Gaffer.ObjectPlug( defaultValue = IECore.NullObject() )
+	processor["defaultCyclesShader"] = Gaffer.ObjectPlug( defaultValue = IECore.NullObject() )
+	processor["defaultDelightShader"] = Gaffer.ObjectPlug( defaultValue = IECore.NullObject() )
+
+	processor["__optionQuery"] = GafferScene.OptionQuery()
+	processor["__optionQuery"]["scene"].setInput( processor["in"] )
+	processor["__optionQuery"].addQuery( Gaffer.ObjectPlug( defaultValue = IECore.NullObject() ) )
+	processor["__optionQuery"].addQuery( Gaffer.ObjectPlug( defaultValue = IECore.NullObject() ) )
+	processor["__optionQuery"]["queries"][0]["name"].setValue( "renderPass:shader:{}:*".format( usage ) )
+
+	processor["__rendererExpression"] = Gaffer.Expression()
+	processor["__rendererExpression"].setExpression(
+		inspect.cleandoc(
+			"""
+			renderer = "3Delight" if parent["renderer"].startswith( "3Delight" ) else parent["renderer"]
```

Comment:
This "3Delight Cloud" thing is seeming less of a good idea with every little hack. If we go the metadata approach for providing default shaders, maybe we can use a wildcarded registration?

Reply:
Yeah, "3Delight Cloud" is proving less elegant to deal with than we first imagined... I've added a `\todo` to remind us about the potential for a wildcarded registration in 1f769f08c7534286e22598436d9465ff2a63598e.